src/algorithms/synthesis/encoding.cpp

```diff
@@ -40,15 +40,23 @@ namespace syrec {
         }
     }
 
-    auto encodeHuffman(TruthTable& tt) -> std::tuple<TruthTable::CubeMap, std::size_t> {
+    auto completeTruthTable(TruthTable& ttExpected, std::size_t const& totalNoBits) -> void {
+        extend(ttExpected);
+        for (auto& [input, output]: ttExpected) {
+            output.resize(totalNoBits);
+        }
+        augmentWithConstants(ttExpected, totalNoBits);
```

Comment:
internally, this again iterates over the TT twice. this is unnecessary and the for loop here can just be included in the `augmentWithConstants` method.
Since then, this function merely consists of two function calls, it might not be needed after all and these function calls can just be placed at the call site.